src/domain/validations.ts

```diff
@@ -109,6 +141,7 @@ export function validateCollaborator(collaborator: Collaborator, application: Ap
       application.sections.applicant.info.primaryAffiliation,
       errors,
     ),
+    compareWithApplicant ? validateNoMatchingApplicant(application, collaborator, errors) : true,
```

Comment:
I would change this flag  `compareWithApplicant` to  `shouldValidateNoMatchingApplicant`